scripts/version_bounds.py

```diff
@@ -0,0 +1,18 @@
+"""Script for functions fetching the minimum versions required by scipy."""
+
+import importlib.metadata
+
+
+def scipy_minimum_python() -> str:
+    """Fetch the minimum Python version specified in pyproject.toml."""
+    raw_version = importlib.metadata.metadata("scipy")["Requires-Python"]
+    if "<" in raw_version:
+        raise NotImplementedError("Version specifier with upper bound not yet supported!")
+    return raw_version.replace(">=", "").replace("~=", "")
+
+
+def scipy_minimum_numpy() -> str:
+    """Fetch the minimum numpy version required by the current scipy package."""
+    np_minimum_dep = next(req for req in importlib.metadata.requires("scipy") if req.startswith("numpy") and " extra " not in req)
+    np_minimum_dep = next(ver for ver in np_minimum_dep.split(",") if ">" in ver)
+    return np_minimum_dep.replace("numpy", "").replace(">=", "").replace(">", "")
```

Comment:
how about something like

```py
def main() -> None:
    import sys

    match sys.argv:
        case [_, "python"]:
            print(scipy_minimum_python())
        case [_, "numpy"]:
            print(scipy_minimum_numpy())
        case _ as badargs:
            print(f"Usage: {badargs[0]} [python|numpy]", file=sys.stderr)
            sys.exit(1)
    

if __name__ == "__main__":
    def main()
```

so that you can do 

```shell
uv sync --frozen --python=$(uv run scripts/version_bounds.py python)
uv pip install --upgrade numpy==$(uv run scripts/version_bounds.py numpy)
```